lib/libspl/include/umem.h

```diff
@@ -137,6 +137,21 @@ umem_free(const void *ptr, size_t size __maybe_unused)
 	free((void *)ptr);
 }
 
+/*
+ * umem_free_aligned was readded for supporting portability
+ * with non-POSIX platforms that require a different free
+ * to be used with aligned allocations.
+ */
+static inline void
+umem_free_aligned(void *ptr, size_t size __maybe_unused)
+{
+#ifndef _WIN32
+	free((void *)ptr);
+#else
+	_aligned_free(ptr);
+#endif
+}
```

Comment:
@behlendorf it's static now
@lundman is this _aligned_free acceptable here or do I need to put it somewhere else?

Reply:
From my perspective it'd be nice to drop the WIN32 case for now since it's dead code until the rest of the windows changes are merged.  That said, I don't feel strongly about it.  We're already carrying some WIN32 cases in the lua and zstd code.

Reply:
Looks fine to me, we didn't want `include/os/$arch/umem.h` ?

Reply:
i didnt see any other changes related to umem in the windows branch so this might be the only change.